src/backend/shared/language.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Supported languages to generate documentation in.
+ */
+export class DocumentationLanguage {
+  /**
+   * TypeScript.
+   */
+  public static readonly TYPESCRIPT = new DocumentationLanguage('typescript');
+
+  /**
+   * Python.
+   */
+  public static readonly PYTHON = new DocumentationLanguage('python');
+
+  /**
+   * Transform a literal string to the `DocumentationLanguage` object.
+   *
+   * Throws an `UnsupportedLanguageError` if the language is not supported.
+   */
+  public static fromString(lang: string) {
+    switch (lang) {
+      case DocumentationLanguage.TYPESCRIPT.toString():
+        return DocumentationLanguage.TYPESCRIPT;
+      case DocumentationLanguage.PYTHON.toString():
+        return DocumentationLanguage.PYTHON;
+      default:
+        throw new UnsupportedLanguageError(lang, [DocumentationLanguage.TYPESCRIPT, DocumentationLanguage.PYTHON]);
+    }
+  }
+
+  private constructor(private readonly lang: string) {}
+
+  public toString() {
+    return this.lang;
+  }
+}
```

Comment:
Why not use an `enum`? I feel it achieves the same outcome with a fraction of the code?

```suggestion
/**
 * Supported languages to generate documentation in.
 */
export const enum DocumentationLanguage {
  /**
   * TypeScript.
   */
  TYPESCRIPT = 'typescript';

  /**
   * Python.
   */
  PYTHON = 'python',
}
```

Reply:
@RomainMuller Yeah, I just wanted to have the `valueOf` functionality so that callers don't have to switch case all the time.

Any suggestion how to achieve that with an `enum`?

Reply:
Not sure I follow... If callers use the enum constants, they don't have to _think_ about case...?

But other than that, if using `export enum` (and not `export const enum`), you can use

```
export function findDocumentationLanguage(lang: string): DocumentationLanguage | undefined {
  return Object.entries(DodumentationLanguage)
    .find(([_, value] => value === lang)?.[0];
}
```